subsys/mgmt/updatehub/updatehub.c

```diff
@@ -355,17 +355,18 @@ static int send_request(enum coap_msgtype msgtype, enum coap_method method,
 	return ret;
 }
 
+#if defined(CONFIG_UPDATEHUB_DOWNLOAD_SHA256_VERIFICATION) || \
+	defined(CONFIG_UPDATEHUB_DOWNLOAD_STORAGE_SHA256_VERIFICATION)
```

Comment:
You are repeating these a lot. Maybe it is worth to define some identifier that will be true if either of these is true.
This may also be done via hidden Kconfig option, that would default to "y" whenever any of other config options get selected.

Reply:
I would prefer to use two switch-define for preprocesor. Something like: `_DOWNLOAD_SHA256_VERIFICATION` and `_STORAGE_SHA256_VERIFICATION` which could be set/clear depends on Kconfig chosen option.
``` C
#ifdef CONFIG_UPDATEHUB_DOWNLOAD_SHA256_VERIFICATION
  #define _DOWNLOAD_SHA256_VERIFICATION
#elif defined(CONFIG_UPDATEHUB_DOWNLOAD_STORAGE_SHA256_VERIFICATION)
  #define _DOWNLOAD_SHA256_VERIFICATION
  #define _STORAGE_SHA256_VERIFICATION
#elif defined(CONFIG_UPDATEHUB_STORAGE_SHA256_VERIFICATION)
  #define _STORAGE_SHA256_VERIFICATION
#endif
```